feat: Add new Saddle pools to 4 chains, plus new tokens.

hammeiam opened the pull request:
<!---
The PR title should follow [conventional commits](https://www.conventionalcommits.org/)
The title will be used to generate the [changelog](/CHANGELOG.md) and release notes, so be descriptive.
You can use prefixes other than fix: and feat: if you think your change should not go in the [changelog](/CHANGELOG.md).
When a new version is released, the API will automatically be deployed to all environments (once a week).
-->

# Description
<!--- Describe your changes in detail -->
- Add new Saddle pools to mainnet
- Create Saddle pool definitions for Fantom, Arbitrum, and Optimism
- Add new tokens to chains where needed


# Testing Instructions

<!--- Please describe how reviewers can test your changes -->
n/a

# Checklist

<!--- The following points should be used to indicate the progress of your PR.  Put an `x` in all the boxes that apply right now, and come back over time and check them off as you make progress.  If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

-   [x] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [x] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

kyu-c reviewed (commented):
LGTM overall!

Left some comments about dry pools. 
Once those are removed, I can do some testing on my end.

hammeiam commented on src/asset-swapper/utils/market_operation_utils/constants.ts:
```diff
@@ -1636,46 +1686,149 @@ export const XSIGMA_MAINNET_INFOS: { [name: string]: CurveInfo } = {
+        pool: SADDLE_MAINNET_POOLS.fraxBP,
+        gasSchedule: 150e3,
+    }),
+    [SADDLE_MAINNET_POOLS.frax3Pool]: createSaddleSwapPool({
```
Done!

hammeiam commented:
@kyu-c made the changes you requested and merged in the latest master 👍

kyu-c commented:
Thanks @hammeiam !! 

I have done testing and it looks good. For Arbitrum though there were some issues with testing (we haven't launched on Arbitrum). My recommendation is to merge the changes for Ethereum and Optimism first and revisit Arbitrum. 

Looks like there is an issue with our CI for an external PR. 

For the interest of time, as it might take a while to fix the CI issue, I created a new PR (#964) with small modifications.

1) Adjusted gas schedule a bit (was underestimating for a few pools)
2) Removed some dead code (e.g. Fantom related from the very first commit)
3) Ran formatter
4) Temporarily removed Arbitrum support (left the metadata there so that it's easier to re-enable in the future)

If you don't mind, I can merge #964 PR instead. If you prefer you can also get the diffs from #964 and merge this PR! Let me know what you prefer. :)

hammeiam commented:
> Thanks @hammeiam !!
> 
> I have done testing and it looks good. For Arbitrum though there were some issues with testing (we haven't launched on Arbitrum). My recommendation is to merge the changes for Ethereum and Optimism first and revisit Arbitrum.
> 
> Looks like there is an issue with our CI for an external PR.
> 
> For the interest of time, as it might take a while to fix the CI issue, I created a new PR (#964) with small modifications.
> 
>     1. Adjusted gas schedule a bit (was underestimating for a few pools)
> 
>     2. Removed some dead code (e.g. Fantom related from the very first commit)
> 
>     3. Ran formatter
> 
>     4. Temporarily removed Arbitrum support (left the metadata there so that it's easier to re-enable in the future)
> 
> 
> If you don't mind, I can merge #964 PR instead. If you prefer you can also get the diffs from #964 and merge this PR! Let me know what you prefer. :)

@kyu-c That works for me, feel free to pull the trigger!

kyu-c commented:
@hammeiam great! I should be able to merge soon. 

Thanks for the contribution! ❤️
